### PR TITLE
ref(environments): Add `Environment.get_for_organization_id` method

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: tests/sentry/models/test_environment.py [hunk 1]
```diff
@@ -22,3 +30,9 @@ def test_simple(self):
         )
 
         assert env2.id == env.id
+
+        with self.assertNumQueries(0):
```
Comment (tkaemming) [inline tests/sentry/models/test_environment.py]: I was going to add this to the second `get_or_create` call but turns out that always tries to do an `INSERT` on `sentry_environmentproject` which wasn't what I was expecting

File: src/sentry/models/environment.py [hunk 2]
```diff
@@ -48,8 +48,28 @@ def get_cache_key(cls, organization_id, name):
         return 'env:2:%s:%s' % (organization_id, md5_text(name).hexdigest())
 
     @classmethod
+    def get_name_or_default(cls, name):
+        return name or ''
```
Comment (tkaemming) [inline src/sentry/models/environment.py]: Not related to this but using an `or` here instead of `is not None` seems odd to me but that's how it is I guess

**Review by tkaemming — COMMENTED** [review]
@bretthoerner and I need this for our environment API work

File: src/sentry/models/environment.py [hunk 3]
```diff
@@ -48,8 +48,28 @@ def get_cache_key(cls, organization_id, name):
...
+
+        cache_key = cls.get_cache_key(organization_id, name)
+
+        env = cache.get(cache_key)
+        if env is None:
+            env = cls.objects.get(
+                name=name,
+                organization_id=organization_id,
+            )
+            cache.set(cache_key, env, 3600)
```
Comment (mattrobenolt) [inline src/sentry/models/environment.py]: I assume we don't ever need to worry about cache invalidation?

Comment (tkaemming) [reply src/sentry/models/environment.py]: Just cargo culting https://github.com/getsentry/sentry/blob/f0814f003bae8e008463d43d34c8b67bb4b65a35/src/sentry/models/environment.py#L82

Comment (mattrobenolt) [reply src/sentry/models/environment.py]: Ah. I see. 🥇

**Review by mattrobenolt — APPROVED** [review]

File: src/sentry/models/environment.py (hunk 3 above)
Comment (tkaemming) [reply src/sentry/models/environment.py]: As far as I know these can't be edited or deleted (but I don't think that's enforced anywhere either)

File: src/sentry/models/environment.py [hunk 4]
```diff
@@ -48,8 +48,28 @@ def get_cache_key(cls, organization_id, name):
...
+
+        return env
+
+    @classmethod
     def get_or_create(cls, project, name):
-        name = name or ''
+        name = cls.get_name_or_default(name)
```
Comment (bretthoerner) [inline src/sentry/models/environment.py]: ❤️ may want to go ahead and use this here? https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/unmerge.py#L321

Comment (tkaemming) [reply src/sentry/models/environment.py]: You've uncovered my not so secret plan to inflate my contribution graph numbers

**Review by bretthoerner — APPROVED** [review]

**Pull request merged**